std/stdio.d

```diff
@@ -1522,6 +1524,143 @@ for every line.
     }
 
 /**
+Creates a file with a randomly generated name and returns an open File to it.
+
+The name will start with the given prefix and be located in the given directory.
+If no directory is given, then it will default to the result of
+$(XREF file, tempDir).
+
+There is nothing special about the file created other than the fact that its
+name is randomly generated, and it defaults to being in the temp directory on
+the system. It will only be deleted if explicitly deleted or if it's in the temp
+directory, and the temp is cleared out (as happens on system startup or shutdown
+on some systems).
+
+Params:
+    prefix = Optional prefix for file name.
+    suffix = Optional suffix for file name (which also provides a way to give
+             the file an extension).
+    dir    = Directory of temporary file.
+  */
+    static File tempFile(string prefix = "", string suffix = "", string dir = "")
+    {
+        import std.exception : ErrnoException;
+        import std.file : isSymlink, tempDir;
+        import std.path : absolutePath, baseName, buildPath, dirName;
+
+        // For some reason, import std.file at the module level of std.stdio or
+        // even just inside of File results in a static assertion being hit in
+        // std.range.primitives.put. So, for now at least, we'll just assign
+        // dir here rather than using tempDir as a default argument so that the
+        // import can be local.
+        if(dir.empty)
+            dir = std.file.tempDir();
```

Comment:
That's even better because default arguments lead to code duplication on every call-site.